src/filesys.cpp

```diff
@@ -740,5 +740,26 @@ bool Rename(const std::string &from, const std::string &to)
 	return rename(from.c_str(), to.c_str()) == 0;
 }
 
+bool bad_char(const char c){
+	if (c >= 0 && c <= 31)
+		return true;
+	if (bad_chars.find_first_of(c) != std::string::npos)
+		return true;
+	if (c > 255)
```

Comment:
c cannot be greater than 255 as you pass a char to the function. The overflow should be handled in caller function